src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+                } else {
+                    this.alertService.error('artemisApp.attachment.pdfPreview.attachmentUnitIDError');
+                }
+            }
+        });
+        document.addEventListener('keydown', this.handleKeyboardEvents);
+        window.addEventListener('resize', this.resizeCanvasBasedOnContainer);
+    }
+
+    ngOnDestroy() {
+        document.removeEventListener('keydown', this.handleKeyboardEvents);
+        window.removeEventListener('resize', this.resizeCanvasBasedOnContainer);
+    }
+
+    private async loadPdf(fileUrl: string) {
+        try {
+            const loadingTask = PDFJS.getDocument(fileUrl);
+            const pdf = await loadingTask.promise;
+            this.totalPages = pdf.numPages;
+
+            for (let i = 1; i <= pdf.numPages; i++) {
+                const page = await pdf.getPage(i);
+                const viewport = page.getViewport({ scale: 1 });
+                const canvas = this.createCanvas(viewport);
+                const context = canvas.getContext('2d');
+                if (context) {
+                    await page.render({ canvasContext: context, viewport }).promise;
+                }
+
+                const container = this.createContainer(canvas, i);
+                this.pdfContainer.nativeElement.appendChild(container);
+            }
+
+            URL.revokeObjectURL(fileUrl);
+        } catch (error) {
+            onError(this.alertService, error);
+        }
+    }
+
+    private createCanvas(viewport: PDFJS.PageViewport): HTMLCanvasElement {
+        const canvas = document.createElement('canvas');
+        canvas.height = viewport.height;
+        canvas.width = viewport.width;
+        const fixedWidth = 250;
+        const scaleFactor = fixedWidth / viewport.width;
+        canvas.style.width = `${fixedWidth}px`;
+        canvas.style.height = `${viewport.height * scaleFactor}px`;
+        return canvas;
+    }
+
+    private createContainer(canvas: HTMLCanvasElement, pageIndex: number): HTMLDivElement {
+        const container = document.createElement('div');
+        container.classList.add('pdf-page-container');
+        container.style.cssText = `position: relative; display: inline-block; width: ${canvas.style.width}; height: ${canvas.style.height}; margin: 20px; box-shadow: 0 2px 6px var(--pdf-preview-canvas-shadow);`;
+
+        const overlay = this.createOverlay(pageIndex);
+        container.appendChild(canvas);
+        container.appendChild(overlay);
+
+        container.addEventListener('mouseenter', () => (overlay.style.opacity = '1'));
+        container.addEventListener('mouseleave', () => (overlay.style.opacity = '0'));
+        overlay.addEventListener('click', () => this.displayEnlargedCanvas(canvas, pageIndex));
+
+        return container;
+    }
+
+    private createOverlay(pageIndex: number): HTMLDivElement {
+        const overlay = document.createElement('div');
+        overlay.classList.add('pdf-page-overlay');
```

Comment:
I couldn't find this class anywhere. Where is this defined?

Reply:
This should be a leftover from when there was a css class, deleted